### PR TITLE
Move SuggestedPodTemplateResources from factory to set_resources

### LINKED ISSUE #64106: Remove unrelated methods from the factory
To continue the work from https://github.com/kubernetes/kubernetes/pull/63984 and https://github.com/kubernetes/kubernetes/pull/63884 this issue helps track additional methods from the cmdutil.Factory that should further be removed, or turned into helpers, independent from the factory.

In *pkg/kubectl/cmd/util/factory_client_access.go*

- [x] MapBasedSelectorForObject `turn into helper` https://github.com/kubernetes/kubernetes/pull/64132
- [x] ProtocolsForObject `turn into helper` https://github.com/kubernetes/kubernetes/pull/64162
- [x] SuggestedPodTemplateResources `move to set_resources.go` https://github.com/kubernetes/kubernetes/pull/64134
- [x] Pauser / Resumer `move to rollout_pause.go / rollout_resume.go command` https://github.com/kubernetes/kubernetes/pull/64121/files
- [x] LabelsForObject `can be completely removed` https://github.com/kubernetes/kubernetes/pull/64113
- [x] Command `turn into helper` https://github.com/kubernetes/kubernetes/pull/64114
- [x] ResolveImage `can be completely removed` https://github.com/kubernetes/kubernetes/pull/64113
- [x] CanBeExposed `move to expose.go` https://github.com/kubernetes/kubernetes/pull/64111

Most of these should be turned into cmd/util helpers under the [polymorphichelpers](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubectl/polymorphichelpers) package with a specific `.go` file for each, and an exposed global variable under [polymorphichelpers/interface.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/polymorphichelpers/interface.go) actually exposing a mutate-able var for that helper (refer to the examples already there).

cc @soltysh 

/assign @ellenkorbes 
/assign @juanvallejo
/assign @deads2k

### PR DESCRIPTION
**Pull request by wgliang**
**What this PR does / why we need it**:

Move SuggestedPodTemplateResources from factory to set_resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #64106

**Special notes for your reviewer**:

/cc @juanvallejo
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/set/set_resources.go [hunk 1]
```diff
@@ -294,3 +294,14 @@ func (o *SetResourcesOptions) Run() error {
 	}
 	return utilerrors.NewAggregate(allErrs)
 }
+
+// SuggestedPodTemplateResources returns a list of resource types that declare a pod template
+func (o *SetResourcesOptions) SuggestedPodTemplateResources() []schema.GroupResource {
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/set/set_resources.go]: this doesn't appear to be used by anything outside of this package - make private

File: pkg/kubectl/cmd/set/set_resources.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func NewCmdResources(f cmdutil.Factory, streams genericclioptions.IOStreams) *co
 	o := NewResourcesOptions(streams)
 
 	resourceTypesWithPodTemplate := []string{}
-	for _, resource := range f.SuggestedPodTemplateResources() {
+	for _, resource := range o.SuggestedPodTemplateResources() {
```
Comment (deads2k) [inline pkg/kubectl/cmd/set/set_resources.go]: The list of resources that contain pod templates is not statically determinant.  Similar to `get` and `describe` and other commands that generic resources, we should suggest the use of `kubectl api-resources` instead.  See `SuggestApiResources` in pkg/kubectl/cmd/util .

Comment (wgliang) [reply pkg/kubectl/cmd/set/set_resources.go]: If so, then I think `suggestedPodTemplateResources` can be deleted. @juanvallejo WDYT?

Comment (juanvallejo) [reply pkg/kubectl/cmd/set/set_resources.go]: @wgliang Yep, I agree it should be deleted

**Comment by soltysh** [comment]
@wgliang will be able to update this within the next 24hours? If not @juanvallejo or I will cherry-pick your changes and update this PR to have it merged before the code slush next week.

**Comment by wgliang** [comment]
@soltysh 
I am ok. I will update it soon.

**Comment by wgliang** [comment]
@juanvallejo @soltysh 
PTAL

**Review by juanvallejo — APPROVED** [review]

**Review by soltysh — APPROVED** [review]

**Pull request merged**